antarest/study/business/timeseries_config_management.py

```diff
@@ -0,0 +1,259 @@
+    "refresh": f"{GENERAL_DATA_PATH}/general/refreshtimeseries",
+    "intra_modal": f"{GENERAL_DATA_PATH}/general/intra-modal",
+    "inter_modal": f"{GENERAL_DATA_PATH}/general/inter-modal",
+    "store_in_input": f"{GENERAL_DATA_PATH}/input/import",
+    "store_in_output": f"{GENERAL_DATA_PATH}/output/archives",
+}
+
+
+class TimeSeriesConfigManager:
+    def __init__(self, storage_service: StudyStorageService) -> None:
+        self.storage_service = storage_service
+
+    def get_ts_field_values(self, study: Study) -> TSFormFields:
+        file_study = self.storage_service.get_storage(study).get_raw(study)
+        general_data = file_study.tree.get(GENERAL_DATA_PATH.split("/"))
+
+        fields = {
+            ts_type: self.__get_form_fields_for_type(
+                file_study, TSType(ts_type), general_data
+            )
+            for ts_type in TSType
+        }
+
+        return TSFormFields(**fields)
+
+    def set_ts_field_values(
+        self, study: Study, field_values: TSFormFields
+    ) -> None:
+        file_study = self.storage_service.get_storage(study).get_raw(study)
+
+        for ts_type, values in field_values:
+            if values is not None:
+                self.__set_ts_field_values_for_type(
+                    study, file_study, TSType(ts_type), values
+                )
+
+    def __set_ts_field_values_for_type(
+        self,
+        study: Study,
+        file_study: FileStudy,
+        ts_type: TSType,
+        field_values: TSFormFieldsForType,
+    ) -> None:
+
+        commands: List[UpdateConfig] = []
+        values = field_values.dict()
+
+        for field, path in PATH_BY_TS_STR_FIELD.items():
+            field_val = values[field]
+            if field_val is not None:
+                commands.append(
+                    self.__set_ts_types_str(
+                        file_study, path, {ts_type: field_val}
+                    )
+                )
+
+        if field_values.number is not None:
+            commands.append(
+                UpdateConfig(
+                    target=f"settings/generaldata/general/nbtimeseries{ts_type}",
+                    data=field_values.number,
+                    command_context=self.storage_service.variant_study_service.command_factory.command_context,
+                )
+            )
+
+        if field_values.refresh_interval is not None:
+            commands.append(
+                UpdateConfig(
+                    target=f"settings/generaldata/general/refreshinterval{ts_type}",
+                    data=field_values.refresh_interval,
+                    command_context=self.storage_service.variant_study_service.command_factory.command_context,
+                )
+            )
+
+        if field_values.season_correlation is not None:
+            commands.append(
+                UpdateConfig(
+                    target=f"settings/generaldata/input/{ts_type}/prepro/correlation/general/mode",
```

Comment:
the path is `input/{ts_type}/prepro/correlation/general/mode`

Reply:
Fixed